website/docs/cairo-1/05-testing/02-integration-testing.md

```diff
@@ -0,0 +1,82 @@
+---
+sidebar_label: Integration testing
+---
+
+# Integration testing
+
+Using [unit testing](./01-unit-testing.md) as much as possible is a good practice, as it makes your test suites run faster. However, when writing smart contracts you often want your test to cover on-chain state and interactions between multiple contracts.
+
+In this section you will learn how to deploy and interact with a smart contract in Protostar for testing purposes. 
+
+## How to test a contract
+To test a contract you need to use important Protostar feature, [cheatcodes](./03-cheatcodes.md). Cheatcodes are additional library functions which Protostar exposes to help you with testing.
+
+Let's write a test which deploys and calls a contract. First let's define our contract 
+
+```cairo title="Deployed contract"
+#[contract]
+mod MinimalContract {
+    #[external]
+    fn panic_with(panic_data: Array::<felt252>) {
+        panic(panic_data);
+    }
+}
+```
+
+We can write a test which deploys and calls this contract. Let's create a file `test_contract.cairo`:
+```cairo title="Example"
+use array::ArrayTrait;
+use result::ResultTrait;
+
+#[test]
+fn test_deploy() {
+    let deployed_contract_address = deploy_contract('minimal', ArrayTrait::new()).unwrap();
+    TODO (add calling code)
```

Comment:
```suggestion
    // TODO (add calling code)
```
They may wanna check if deploying itself works so we might not want to break compilation

Reply:
I will add this code here in this PR